custom_components/waste_collection_schedule/waste_collection_schedule/source/bep_environnement_be.py

```diff
@@ -0,0 +1,126 @@
+    # Papiers cartons
+    "papierscartons": {"type": "Papiers & Cartons", "icon": "mdi:leaf"}
+}
+
+#### Arguments affecting the configuration GUI ####
+
+HOW_TO_GET_ARGUMENTS_DESCRIPTION = {  # Optional dictionary to describe how to get the arguments, will be shown in the GUI configuration form above the input fields, does not need to be translated in all languages
+    "en": "HOW TO GET ARGUMENTS DESCRIPTION",
+}
+
+PARAM_DESCRIPTIONS = {  # Optional dict to describe the arguments, will be shown in the GUI configuration below the respective input field
+    "en": {
+        "locality": "Name of the locality"
+    }
+}
+
+
+def GetCitiesValue() -> List[dict]:
+    """Return id for each city available in calendar
+
+    Returns:
+        List[dict]: key is the city, value is the id for the calendar
+    """
+    response = requests.get(URL)
+    soup = BeautifulSoup(response.text, 'html.parser')
+
+    # Find all <option> elements inside the select
+    options = soup.select('#locform-loc option')
+
+    # Create a dictionary of city names and their values
+    city_values = {option.text.lower(): option['value']
+                   for option in options if option.text.strip()}
+
+    # Print the result
+    return city_values
+
+
+def BepWasteParser(response: dict) -> List[Collection]:
+    """Specific implementation to parse response from Bep-Environement
+
+    Args:
+        response (dict): response from the website
+
+    Returns:
+        List[Collection]: list of Collection found
+    """
+    collections = []
+    cal_soup = BeautifulSoup(response.get("cal", ""), "html.parser")
+    cells = cal_soup.find_all("td")
+
+    for cell in cells:
+        # Extract attributes
+        data_day = cell.get("data-day")
+        data_month = cell.get("data-month")
+        data_year = cell.get("data-year")
+
+        # Check for waste types in the class attribute
+        classes = cell.get("class", [])
+        waste_types = [cls for cls in classes if cls in WASTE_MAP]
+
+        if data_day and data_month and data_year and waste_types:
+            # Construct the date
+            collection_date = date(
+                year=int(data_year), month=int(data_month), day=int(data_day)
+            )
+
+            for abbr, map in WASTE_MAP.items():
+                if abbr in waste_types:
+                    c = Collection(date=collection_date,
+                                   t=map["type"], icon=map["icon"])
+                    collections.append(c)
+
+    return collections
+
+
+class Source:
+    BEP_CALENDAR_URL = "https://www.bep-environnement.be/wp-admin/admin-ajax.php"
+    GARBAGE_COLLECTION_ACTION = "calendriercollectes"
+
+    def __init__(self, locality: str):
+        self._locality = locality.lower()
+
+    def fetch(self) -> list[Collection]:
+        # Check for city name given
+        citiesValue = GetCitiesValue()
+
+        if self._locality not in citiesValue:
+            raise SourceArgumentException(
```

Comment:
Use `SourceArgumentNotFoundWithSuggestions` here

https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md#exceptions

Reply:
Done, thank you for the review